docs/ml-prediction.md

```diff
@@ -0,0 +1,258 @@
+# GPU based, batch and streaming predictions on Dataflow
+
+## Introduction
+
+Dataflow allows you to run a process:
+
+* in a user-defined Docker container
+* on machines with GPU
+
+These 2 features bring new possibilities for people who want to run batch and streaming
+predictions on Dataflow. Thanks to Dockerized environment, you can define any
+prediction environment you need, and make predictions more efficient using GPU.
+
+BigFlow 1.3 comes with utilities that make your life easier when it comes to writing Dockerized
+processes and writing prediction logic.
+
+This recipe will show you how to set up a PyTorch prediction process, but the dockerization allows you
+to define an environment for any framework. We assume that you are already familiar with the BigFlow API.
+
+We are going to create a batch prediction process for the Fashion MNIST model, using Dataflow. You can download
+the model here. We won't focus on the utility or performance side of the project, but more on the API that
+BigFlow provides, which makes creating prediction processes easier.
+
+## Docker image and packages
+
+To run a prediction process on GPU, you need a Docker file with the CUDA drivers. The image below has all the dependencies
+for running a PyTorch prediction process, and also it's compatible with BigFlow. Use that image as the image for your
+BigFlow project.
+
+```dockerfile
+# Docker base image, includes os-level tools (shell, utilities, services), python interpreter, C-libraries, etc.
+
+# use it if you want ot use custom Docker image on Dataflow workers
+FROM apache/beam_python3.7_sdk:2.27.0
+
+
+# Cherry-pick apache beam binaries from offical image to our custom.
+# Use this only in case when you unable to use original image (FROM apache/beam...)
+# COPY --from=apache/beam_python3.6_sdk:2.26.0 /opt/apache/beam /opt/apache/beam
+
+
+# Uncomment if you want to install CUDA drivers to your docker
```

Comment:
We should reverse this comment imo. `Comment if you do not want to install CUDA drivers...`

Reply:
done, kinda